Master  does not build on OSX

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### Informations

- **Qiskit Aer version**: master
- **Python version**: Py37
- **Operating system**: OSX

### What is the current behavior?

ConanException: Error 127 while executing cd '/Users/paul/.conan/data/muparserx/4.0.8/_/_/build/eec6acc43f6348a597c20e5bd28d9e0590a02597' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Release" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="apple-clang" -DCONAN_COMPILER_VERSION="11.0" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libc++" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/Users/paul/.conan/data/muparserx/4.0.8/_/_/package/eec6acc43f6348a597c20e5bd28d9e0590a02597" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -DBUILD_EXAMPLES="False" -Wno-dev '/Users/paul/.conan/data/muparserx/4.0.8/_/_/build/eec6acc43f6348a597c20e5bd28d9e0590a02597'
CMake Error at cmake/conan.cmake:402 (message):
  Conan install failed='1'
Call Stack (most recent call first):
  cmake/conan.cmake:497 (conan_cmake_install)
  cmake/conan_utils.cmake:37 (conan_cmake_run)
  CMakeLists.txt:117 (setup_conan)


```
Traceback (most recent call last):
  File "/opt/miniconda3/lib/python3.7/site-packages/skbuild/setuptools_wrap.py", line 574, in setup
    languages=cmake_languages
  File "/opt/miniconda3/lib/python3.7/site-packages/skbuild/cmaker.py", line 232, in configure
    os.path.abspath(CMAKE_BUILD_DIR())))

An error occurred while configuring with CMake.
  Command:
    "/Users/paul/Downloads/qiskit-aer-master/.eggs/cmake-3.17.2-py3.7-macosx-10.9-x86_64.egg/cmake/data/CMake.app/Contents/bin/cmake" "/Users/paul/Downloads/qiskit-aer-master" "-G" "Unix Makefiles" "-DCMAKE_INSTALL_PREFIX:PATH=/Users/paul/Downloads/qiskit-aer-master/_skbuild/macosx-10.9-x86_64-3.7/cmake-install" "-DPYTHON_EXECUTABLE:FILEPATH=/opt/miniconda3/bin/python" "-DPYTHON_VERSION_STRING:STRING=3.7.7" "-DPYTHON_INCLUDE_DIR:PATH=/opt/miniconda3/include/python3.7m" "-DPYTHON_LIBRARY:FILEPATH=/opt/miniconda3/lib/libpython3.7m.dylib" "-DSKBUILD:BOOL=TRUE" "-DCMAKE_MODULE_PATH:PATH=/opt/miniconda3/lib/python3.7/site-packages/skbuild/resources/cmake" "-DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=10.9" "-DCMAKE_BUILD_TYPE:STRING=Release" "-DCMAKE_OSX_ARCHITECTURES:STRING=x86_64"
  Source directory:
    /Users/paul/Downloads/qiskit-aer-master
  Working directory:
    /Users/paul/Downloads/qiskit-aer-master/_skbuild/macosx-10.9-x86_64-3.7/cmake-build
```


### Steps to reproduce the problem



### What is the expected behavior?



### Suggested solutions

--- a/cmake/conan_utils.cmake
+++ b/cmake/conan_utils.cmake
@@ -34,5 +34,11 @@ macro(setup_conan)
         set(REQUIREMENTS ${REQUIREMENTS} catch2/2.12.1)
     endif()
 
-    conan_cmake_run(REQUIRES ${REQUIREMENTS} OPTIONS ${CONAN_OPTIONS} BASIC_SETUP CMAKE_TARGETS KEEP_RPATHS BUILD missing)
+    conan_cmake_run(REQUIRES ${REQUIREMENTS}
+                    OPTIONS ${CONAN_OPTIONS}
+                    ENV CONAN_CMAKE_PROGRAM=${CMAKE_COMMAND}
+                    BASIC_SETUP
+                    CMAKE_TARGETS
+                    KEEP_RPATHS
+                    BUILD missing)
 endmacro()

